controllers/topology/conditions.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/scope"
+	"sigs.k8s.io/cluster-api/util/conditions"
+)
+
+func (r *ClusterReconciler) reconcileConditions(s *scope.Scope, cluster *clusterv1.Cluster, reconcileErr error) error {
+	return r.reconcileTopologyReconciledCondition(s, cluster, reconcileErr)
+}
+
+// reconcileTopologyReconciledCondition sets the TopologyReconciled condition on the cluster.
+// The TopologyReconciled condition is considered true is spec of all the objects associated with the
+// cluster are in sync with the topology defined in the cluster.
+// The condition is false under the following conditions:
+// - An error occurred during the reonciled process of the cluster topology.
+// - The cluster upgrade has not yet propogated to all the components of the cluster.
+//   - For a managed topology cluster the version upgrade is propagated one component at a time.
+//     In such a case, since some of the component's spec would be adrift from the topology the
+//     topology cannot be considered fully reconciled.
```

Comment:
```suggestion
// reconcileTopologyReconciledCondition sets the TopologyReconciled condition on the cluster.
// The TopologyReconciled condition is considered true if spec of all the objects associated with the
// cluster are in sync with the topology defined in the cluster.
// The condition is false under the following conditions:
// - An error occurred during the reconcile process of the cluster topology.
// - The cluster upgrade has not yet propagated to all the components of the cluster.
// - For a managed topology cluster the version upgrade is propagated one component at a time.
//   In such a case, since some of the component's spec would be adrift from the topology the
//   topology cannot be considered fully reconciled.
```
nit